File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/model/mapper/UserMapper.java
package com.lattis.ellipse.data.network.model.mapper;

import androidx.annotation.NonNull;

import com.lattis.ellipse.data.network.base.AbstractDataMapper;
import com.lattis.ellipse.data.network.model.response.GetUserResponse;
import com.lattis.ellipse.data.network.model.response.PrivateNetworkResponse;
import com.lattis.ellipse.data.network.model.response.UserResponse;
import com.lattis.ellipse.domain.model.PrivateNetwork;
import com.lattis.ellipse.domain.model.User;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

public class UserMapper extends AbstractDataMapper<GetUserResponse,User> {

    @Inject
    public UserMapper() {}

    @NonNull
    @Override
    public User mapIn(@NonNull GetUserResponse getUserResponse) {
        User user = new User();
        UserResponse userResponse = getUserResponse.getUserResponse();
        user.setId(userResponse.getUserId());
        user.setUsersId(userResponse.getUsersId());
        user.setUsername(userResponse.getUsername());
        user.setVerified(true);
        user.setMaxLocks(userResponse.getMaxLocks());
        user.setTitle(userResponse.getTitle());
        user.setFirstName(userResponse.getFirstName());
        user.setLastName(userResponse.getLastName());
        user.setPhoneNumber(userResponse.getPhoneNumber());
        user.setEmail(userResponse.getEmail());
        List<PrivateNetworkResponse> privateNetworksResponse = getUserResponse.getPrivateNetworkResponse();
        List<PrivateNetwork>privateNetworks = new ArrayList<>();
        if(privateNetworksResponse!=null){
            for(PrivateNetworkResponse privateNetworkResponse: privateNetworksResponse){
                if(privateNetworkResponse!=null){
                    PrivateNetwork privateNetwork = new PrivateNetwork();
                    privateNetwork.setPrivate_fleet_user_id(privateNetworkResponse.getPrivate_fleet_user_id());
                    privateNetwork.setUserId(privateNetworkResponse.getUser_id());
                    privateNetwork.setEmail(privateNetworkResponse.getEmail());
                    privateNetwork.setFleet_id(privateNetworkResponse.getFleet_id());
                    privateNetwork.setVerified(privateNetworkResponse.getVerified());
                    privateNetwork.setFleet_name(privateNetworkResponse.getFleet_name());
                    privateNetwork.setType(privateNetworkResponse.getType());
                    privateNetwork.setLogo(privateNetworkResponse.getLogo());
                    privateNetworks.add(privateNetwork);

                }
            }
        }

        user.setPrivateNetworks(privateNetworks);

        return user;
    }

    @NonNull
    @Override
    public GetUserResponse mapOut(@NonNull User user) {
        return null;
    }
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/mapper/UserMapper.kt
package com.lattis.data.mapper



import com.lattis.data.entity.response.user.GetUserResponse
import com.lattis.domain.models.PrivateNetwork
import com.lattis.domain.models.User

import java.util.ArrayList

import javax.inject.Inject

class UserMapper @Inject
constructor() : AbstractDataMapper<GetUserResponse, User>() {

    override fun mapIn(getUserResponse: GetUserResponse?): User {
        val user = User()

        if(getUserResponse!=null) {
            val userResponse = getUserResponse.getUserPayload()
            user.id = userResponse?.userId
            user.usersId = userResponse?.usersId
            user.username = userResponse?.username
            user.isVerified = userResponse?.isVerified
            user.maxLocks = userResponse?.maxLocks?:0
            user.title = userResponse?.title
            user.firstName = userResponse?.firstName
            user.lastName = userResponse?.lastName
            user.phoneNumber = userResponse?.phoneNumber
            user.email = userResponse?.email
            user.restToken = userResponse?.restToken
            user.refreshToken = userResponse?.refreshToken
            user.userType = userResponse?.userType

            val privateNetworksResponse = getUserResponse.privateNetworkResponse
            val privateNetworks = ArrayList<PrivateNetwork>()
            if (privateNetworksResponse != null) {
                for (privateNetworkResponse in privateNetworksResponse) {
                    if (privateNetworkResponse != null) {
                        val privateNetwork = PrivateNetwork()
                        privateNetwork.private_fleet_user_id = privateNetworkResponse.private_fleet_user_id
                        privateNetwork.user_id = privateNetworkResponse.user_id
                        privateNetwork.email = privateNetworkResponse.email
                        privateNetwork.fleet_id = privateNetworkResponse.fleet_id
                        privateNetwork.verified = privateNetworkResponse.verified
                        privateNetwork.fleet_name = privateNetworkResponse.fleet_name
                        privateNetwork.type = privateNetworkResponse.type
                        privateNetwork.logo = privateNetworkResponse.logo

                        if(privateNetworkResponse.address!=null){
                            val address = PrivateNetwork.Address()
                            address?.address_id = privateNetworkResponse.address.address_id
                            address?.address1 = privateNetworkResponse.address.address1
                            address?.address2 = privateNetworkResponse.address.address2
                            address?.city = privateNetworkResponse.address.city
                            address?.country = privateNetworkResponse.address.country
                            address?.postal_code = privateNetworkResponse.address.postal_code
                            address?.state = privateNetworkResponse.address.postal_code
                            address?.type = privateNetworkResponse.address.type
                            address?.type_id = privateNetworkResponse.address.type_id
                            privateNetwork.address = address
                        }
                        privateNetworks.add(privateNetwork)

                    }
                }
            }

            user.privateNetworks = privateNetworks
        }

        return user
    }

    override fun mapOut(user: User?): GetUserResponse? {
        return null
    }
}
